payments-core/src/main/java/com/stripe/android/googlepaylauncher/GooglePayPaymentMethodLauncherViewModel.kt

```diff
@@ -0,0 +1,169 @@
+    internal val googlePayResult = _googleResult.distinctUntilChanged()
+
+    fun updateResult(result: GooglePayPaymentMethodLauncher.Result) {
+        _googleResult.value = result
+    }
+
+    @VisibleForTesting
+    suspend fun isReadyToPay(): Boolean {
+        return googlePayRepository.isReady().first()
+    }
+
+    @VisibleForTesting
+    fun createPaymentDataRequest(): JSONObject {
+        return googlePayJsonFactory.createPaymentDataRequest(
+            transactionInfo = createTransactionInfo(args),
+            merchantInfo = GooglePayJsonFactory.MerchantInfo(
+                merchantName = args.config.merchantName
+            ),
+            billingAddressParameters = GooglePayJsonFactory.BillingAddressParameters(
+                isRequired = true,
+                format = GooglePayJsonFactory.BillingAddressParameters.Format.Min,
+                isPhoneNumberRequired = false
+            ),
+            isEmailRequired = args.config.isEmailRequired
+        )
+    }
+
+    @VisibleForTesting
+    internal fun createTransactionInfo(
+        args: GooglePayPaymentMethodLauncherContract.Args
+    ): GooglePayJsonFactory.TransactionInfo {
+        return GooglePayJsonFactory.TransactionInfo(
+            currencyCode = args.currencyCode,
+            totalPriceStatus = when (args.hasAmount) {
+                true -> GooglePayJsonFactory.TransactionInfo.TotalPriceStatus.Final
+
+                false -> GooglePayJsonFactory.TransactionInfo.TotalPriceStatus.NotCurrentlyKnown
+            },
+            countryCode = args.config.merchantCountryCode,
+            totalPrice = args.amount,
+            checkoutOption = GooglePayJsonFactory.TransactionInfo.CheckoutOption.Default
+        )
+    }
+
+    suspend fun createLoadPaymentDataTask(): Task<PaymentData> {
+        check(isReadyToPay()) {
+            "Google Pay is unavailable."
+        }
+        return paymentsClient.loadPaymentData(
+            PaymentDataRequest.fromJson(createPaymentDataRequest().toString())
+        )
+    }
+
+    suspend fun createPaymentMethod(
+        paymentData: PaymentData
+    ): GooglePayPaymentMethodLauncher.Result {
+        val paymentDataJson = JSONObject(paymentData.toJson())
+
+        val params = PaymentMethodCreateParams.createFromGooglePay(paymentDataJson)
+
+        return runCatching {
+            requireNotNull(
+                stripeRepository.createPaymentMethod(params, requestOptions)
+            )
+        }.fold(
+            onSuccess = {
+                GooglePayPaymentMethodLauncher.Result.Completed(it)
+            },
+            onFailure = {
+                GooglePayPaymentMethodLauncher.Result.Failed(it)
+            }
+        )
+    }
+
+    internal class Factory(
+        private val application: Application,
+        private val args: GooglePayPaymentMethodLauncherContract.Args,
+        private val enableLogging: Boolean = false,
+        private val workContext: CoroutineContext = Dispatchers.IO
```

Comment:
enableLogging is always false, and workContext is always Dispatchers.IO.   You can probably make them member variables or remove them.

Reply:
I'll leave them here in case we want to test `Factory`